protob/messages.proto

```diff
@@ -560,6 +584,47 @@ message TxAck {
 	optional TransactionType tx = 1;
 }
 
+/**
+ * Request: Ask device to sign transaction
+ * All fields are optional from the protocol's point of view. Each field defaults to value `0` if missing.
+ * @next PassphraseRequest
+ * @next PinMatrixRequest
+ * @next EthereumTxRequest
+ * @next Failure
+ */
+message EthereumSignTx {
+	repeated uint32 address_n = 1;			// BIP-32 path to derive the key from master node
+	optional bytes nonce = 2;			// 256 bit unsigned big endian
+	optional bytes gas_price = 3;			// 256 bit unsigned big endian (in wei)
+	optional bytes gas_limit = 4;			// 256 bit unsigned big endian
+	optional bytes to = 5;				// 160 bit address hash
+	optional bytes value = 6;			// 256 bit unsigned big endian (in wei)
+	optional uint32 data_length = 7;		// Length of transaction payload
+}
+
+/**
+ * Response: Device asks for more data from transaction payload, or returns the signature.
+ * If data_length is set, device awaits that many more bytes of payload.
+ * Otherwise, the signature_* fields contain the computed transaction signature.
+ * @prev EthereumSignTx
+ * @next EthereumTxAck
+ */
+message EthereumTxRequest {
+	optional uint32 data_length = 1;		// Number of bytes being requested
+	optional uint32 signature_v = 2;		// Computed signature (recovery parameter, limited to 27 or 28)
```

Comment:
We already fix the signature to get `s <= n/2`.  There is also a restriction in Bitcoin; otherwise the transaction is non-standard and not relayed by all nodes (if I remember correctly).